collectors/net.go

```diff
@@ -12,19 +12,35 @@ import (
 
 // Net collects network related metrics
 type Net struct {
-	counters []net.IOCountersStat
-
-	mutex     sync.RWMutex
-	sensision bytes.Buffer
-	level     uint8
-	period    uint
+	counters   []net.IOCountersStat
+	interfaces []string
+	mutex      sync.RWMutex
+	sensision  bytes.Buffer
+	level      uint8
+	period     uint
 }
 
 // NewNet returns an initialized Net collector.
-func NewNet(period uint, level uint8) *Net {
+func NewNet(period uint, level uint8, opts interface{}) *Net {
+
+	var options map[string]interface{}
+	var ifaces []string
+
+	if opts != nil {
+		options = opts.(map[string]interface{})
```

Comment:
opts values are not strings, they are map[string]string.

net-opts is a map[string]map[string]interface{}
in the case of interfaces it's a map[string]map[string][]string.

Hence the use of map[string]interface{} to abstract the properties content.